x-pack/plugins/integration_assistant/server/graphs/log_type_detection/csv.ts

```diff
@@ -0,0 +1,201 @@
+  }
+  const safeName = columnName.replace(/[^a-zA-Z0-9_]/g, '_');
+  return /^[0-9]/.test(safeName) ? `Column${safeName}` : safeName;
+}
+
+// Returns the column list from a header row. We skip values that are not strings.
+export function columnsFromHeader(
+  tempColumnNames: string[],
+  headerObject: { [key: string]: unknown }
+): Array<string | undefined> {
+  const maxIndex = tempColumnNames.findLastIndex(
+    (columnName) => headerObject[columnName] !== undefined
+  );
+  return tempColumnNames
+    .slice(0, maxIndex + 1)
+    .map((columnName) => headerObject[columnName])
+    .map(toSafeColumnName);
+}
+
+// Count the number of columns actually present in the rows.
+export function totalColumnCount(
+  tempColumnNames: string[],
+  csvRows: Array<{ [key: string]: unknown }>
+): number {
+  return (
+    Math.max(
+      -1,
+      ...csvRows.map((row) =>
+        tempColumnNames.findLastIndex((columnName) => row[columnName] !== undefined)
+      )
+    ) + 1
+  );
+}
+
+// Prefixes each column with the provided prefixes, separated by a period.
+function prefixColumns(columns: string[], prefixes: string[]): string[] {
+  return columns.map((column) => [...prefixes, column].join('.'));
+}
+
+/**
+ * Generates a list of unique column names based on preferred and fallback names.
+ *
+ * The preferred names are used first, followed by the fallback names. It is required that
+ * there are enough fallback names to cover the number of unique column names needed.
+ *
+ * The resulting column names are guaranteed to be unique. If a column name is already in use,
+ * a postfix like _2, _3 and so on is added to the name to make it unique.
+ *
+ * @generator
+ * @param {number} count - The number of unique column names to generate.
+ * @param {Array<Array<string | undefined>>} preferredNames - A 2D array where each sub-array contains a list of names.
+ * @param {string[]} fallbackNames - An array of fallback names to use if no preferred name is defined.
+ * @yields {string} - A sequence of column names, such that no two are the same.
+ */
+export function* yieldUniqueColumnNames(
+  count: number,
+  preferredNames: Array<Array<string | undefined>>,
+  fallbackNames: string[]
+): Generator<string, void> {
+  const knownNames = new Set<string>();
+
+  for (let i = 0; i < count; i++) {
+    let selectedName: string = fallbackNames[i];
+
+    for (const nameList of preferredNames) {
+      const name = nameList[i];
+      if (name) {
+        selectedName = name;
+        break;
+      }
+    }
+
+    let postfixString = '';
+
+    if (knownNames.has(selectedName)) {
+      for (let postfix = 2; ; postfix++) {
+        postfixString = `_${postfix}`;
+        if (!knownNames.has(selectedName + postfixString)) {
+          break;
+        }
+      }
+    }
+
+    selectedName += postfixString;
+    knownNames.add(selectedName);
+    yield selectedName;
+  }
+}
+
+// Converts CSV samples into JSON samples.
+export async function handleCSV({
```

Comment:
Extract this into an own csv folder under graphs for any csv related stuff. Probably move all the csv related files in there

Reply:
There's only one file for now 🤷

Reply:
But we have maintained the structure in such a way that graphs are separated and logic is separated.
This file is kind of bloated already

Reply:
Makes sense. I've created the folder and split the file into two.